### PR TITLE
Suggest using double colon when a struct field type include single colon

### PR DESCRIPTION
**Pull request by TaKO8Ki**
#92685

### CODE REVIEW

File: compiler/rustc_parse/src/parser/item.rs [hunk 1]
```diff
@@ -1534,6 +1534,16 @@ impl<'a> Parser<'a> {
         let name = self.parse_field_ident(adt_ty, lo)?;
         self.expect_field_ty_separator()?;
         let ty = self.parse_ty()?;
+        if self.token.kind == token::Colon && self.look_ahead(1, |tok| tok.kind != token::Colon) {
+            self.struct_span_err(self.token.span, "found single colon in a struct field type path")
+                .span_suggestion_verbose(
+                    self.token.span,
+                    "maybe you meant to write a path separator here",
```
Comment (cjgillot) [inline compiler/rustc_parse/src/parser/item.rs:1541]:
```suggestion
                    "write a path separator here",
```

File: src/test/ui/suggestions/sturct-field-type-including-single-colon.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+mod foo {
```
Comment (cjgillot) [inline src/test/ui/suggestions/sturct-field-type-including-single-colon.rs:1]: This file's name has a typo.

**Review by cjgillot — COMMENTED** [review]
Thanks @TaKO8Ki!
I left a few nits. I'll approve once they are resolved.

File: src/test/ui/suggestions/sturct-field-type-including-single-colon.rs (hunk 2 above)
Comment (TaKO8Ki) [reply src/test/ui/suggestions/sturct-field-type-including-single-colon.rs]: I fixed a typo!

**Pull request merged**